rdr_service/genomic/genomic_job_components.py

```
@@ -557,12 +557,18 @@ def _process_gc_metrics_data_for_insert(self, data_to_ingest):
             row_copy['file_id'] = self.file_obj.id
             sample_id = row_copy['sampleid']
 
-            # TODO: limit call rate to 10 characters for now until clarification from GCs
             try:
                 row_copy['callrate'] = row_copy['callrate'][:10]
             except KeyError:
                 pass
 
+            # Clean up contamination if data is less than 0
+            try:
+                if '-' in row_copy['contamination']:
```

Is contamination previously validated to be a number? If not, then there's a chance that this could silently hide that contamination values are invalid. Like if someone decided to try to use a range of "0.02 - 0.05". Definitely an edge case, but this might ignore it and zero it out.

---

Contamination is not previously validated to be a number, but it probably should be. I will make the update.

---

I updated this to validate that `contamination` is a number. This will ensure we can perform the contamination category calculations. I added a TODO in the RDR model to update the datatype to a float. I did not perform this yet as it will impact both the RDR and PDR.